### ISSUE TITLE
[Bug] Android: malformed empty multipart requests

### ISSUE THREAD

**Issue by silviogutierrez** [silkimen/cordova-plugin-advanced-http#372]
**Describe the bug**

First of all: thanks for all your work on this library. It's a life saver with cookies and avoiding CORS.

On Android only, if you post multipart with an empty FormData object, the boundary is not set. This is invalid and some frameworks will refuses to parse the request. See here: https://stackoverflow.com/questions/34326150/multipartparsererror-invalid-boundary

Note: only happens on Android, not on iOS. I don't know much Android native dev, but I added a few breakpoints and `startPart` in HttpRequest.java is not called for empty requests. So the boundary is missing. See screenshots.

**System info**
 - affected HTTP plugin version: all versions higher than and including 2.3.1
 - affected platform(s) and version(s): Android
 - affected device(s): all

**Minimum viable code to reproduce**

Using ionic's wrapper:

```typescript
await HTTP.sendRequest(url, {
  method: "post",
  serializer: "multipart",
  data: new FormData(),
})
```

**Screenshots**
With data:

<img width="676" alt="image" src="https://user-images.githubusercontent.com/92824/94876616-30dd2600-0426-11eb-9e60-519c9a5ae858.png">

Without data (bug):
<img width="552" alt="image" src="https://user-images.githubusercontent.com/92824/94876627-38043400-0426-11eb-8d1d-c7288564f170.png">

**Comment by silkimen** [comment]
Hi Silvio, I've checked how an empty FormData instance is handled in AFNEtworking (on iOS). It's just falling back to an empty url encoded request instead, which seemed strange at first sight. But on the other hand: it never caused problems as yet and sending empty multipart requests should be an extremely rare case. So I've decided to match this behavior on Android. This issue will be fixed in 3.1.1. 🤗
